browser/android/preferences/background_video_playback_tab_helper.cc

```diff
@@ -80,4 +84,74 @@ void BackgroundVideoPlaybackTabHelper::DidFinishNavigation(
   }
 }
 
+namespace chrome {
+namespace android {
+
+mojo::AssociatedRemote<script_injector::mojom::ScriptInjector> GetRemote(
+    content::RenderFrameHost* rfh) {
+  mojo::AssociatedRemote<script_injector::mojom::ScriptInjector>
+      script_injector_remote;
+  rfh->GetRemoteAssociatedInterfaces()->GetInterface(&script_injector_remote);
+  return script_injector_remote;
+}
+void JNI_BackgroundVideoPlaybackTabHelper_SendOrientationChangeEvent(
+    JNIEnv* env,
+    const base::android::JavaParamRef<jobject>& jweb_contents,
+    jboolean is_full_screen) {
+  content::WebContents* web_contents =
+      content::WebContents::FromJavaWebContents(jweb_contents);
+  // Injecting this script to make youtube video fullscreen on landscape mode
+  // and exit fullscreen on portrait mode.
+  std::string script;
+  if (is_full_screen) {
+    script =
+        "if(!document.fullscreenElement) {"
```

Comment:
If YT updates their site and these classes/ids no longer match then Brave is going to look broken.

Reply:
also please assign this to `constexpr const char16_t` and the if/else will just decide which one to run

Reply:
use `uR"js( ... )js"` for multiline scripts in general, but I think we need to source this script from a component to handle YT site changes

Reply:
actually do we even need this script at all? Can't we just call `Browser::EnterFullscreenModeForTab`?

Reply:
sorry, can't use `Browser` here for android. On desktop setting it to fullscreen isn't the same as changing the video to fullscreen so not sure if we can use native methods here or not, but we definitely can below to exit

Reply:
looks like `FullscreenManager` is the equivalent in java, but not sure if it has the desired behavior here or not